Sources/HummingbirdXCT/TestApplication.swift

```diff
@@ -35,20 +35,23 @@ struct TestApplication<BaseApp: HBApplicationProtocol>: HBApplicationProtocol, S
         self.base.server
     }
 
-    /// event loop group used by application
+    /// Event loop group used by application
     var eventLoopGroup: EventLoopGroup { self.base.eventLoopGroup }
     /// Configuration
     var configuration: HBApplicationConfiguration { self.base.configuration.with(address: .hostname("localhost", port: 0)) }
     /// Logger
     var logger: Logger { self.base.logger }
-    /// on server running
+    /// On server running
     @Sendable func onServerRunning(_ channel: Channel) async {
         await self.portPromise.complete(channel.localAddress!.port!)
     }
 
-    /// services attached to the application.
+    /// Services attached to the application.
     var services: [any Service] { self.base.services }
 
+    /// Processes run before server start
+    public var processesRunBeforeServerStart: [@Sendable () async throws -> Void] { self.base.processesRunBeforeServerStart }
```

Comment:
I don't like this variable name, but I don't know a better one.

Reply:
I guess we could use `pre`. Like `preStartProcesses`. A little bit more vague than `processesRunBeforeServerStart`, but should do the job.

In that case, there is also the `runBeforeServerStart` function that needs to have a change of name to something like `preStartRun`. Not sure if these are the best options, but they sound good to me.

Reply:
Yeah it's not a great variable name, but I don't think @MahdiBM your suggestions are much better especially the function rename. `preStartRun` doesn't mean a great deal.

Reply:
Gonna add a Bikeshedding issue with all the symbol names that need revisiting and add this to it

Reply:
#343